package.json

```diff
@@ -57,13 +58,15 @@
     "@folio/stripes-logger": "~1.0.0",
     "classnames": "^2.2.6",
     "final-form": "^4.18.2",
+    "final-form-arrays": "^3.0.2",
```

Comment:
You don't appear to be using final-form-arrays or RFFA.

Which makes me ask...why not use final-form rather than hand-rolling your own form logic and all that in `CustomFieldsForm`? Was there a limitation you were running into? Kind of curious.

Reply:
I agree. It's definitely not perfect to invent the form from scratch instead of using existing solutions. However, edit custom fields page is a bit special and has more complicated logic than other forms. If I used final-form, I would still have to maintain my own state and it's logic for things like storing removed fields, inserting them back, tracking fields that are not yet saved in the back end. It's a bit more tricky than:
1. click submit
2. get the values from the form state 
3. send these values to the back-end

It's not good to have self-written form logic, so eventually  `CustomFieldsForm` will be refactored to utilize `react-final-form`. There is a story for it:  https://issues.folio.org/browse/STSMACOM-299.
This integration can be tricky, so, for now, I use my own logic to deliver the feature in a reasonable time